controllers/finance_controller.py

```diff
@@ -1,26 +1,103 @@
-from tracker.finance_tracker import FinanceTracker
 from datetime import datetime
 
+from sqlalchemy import func, extract, and_
+from sqlalchemy.exc import IntegrityError
 
-class FinanceController:
+from database import SessionLocal
+from models.expense import Expense
+from models.investment import Investment
+from views.utilities import validate_date, validate_and_convert_date
+
+
+class FinanceTracker:
     def __init__(self):
-        self.finance_tracker = FinanceTracker()
+        # Initialize a new SQLAlchemy session
+        self.db_session = SessionLocal()
 
-    def add_expense(self, user_id, category, amount, date_str):
-        success, message = self.finance_tracker.add_expense(user_id, category, amount, date_str)
-        return success, message
+    def add_expense(self, user_id, category, amount, date):
+        # First, validate the date format
+        date_obj, message = validate_and_convert_date(date)
```

Comment:
validate_and_convert_date(date) seems complex, along with the next 20-23 lines.

there should be a fix format that can be dynamic from settings.py os.env var, and raise error by itself if date is wrongly formatted.

you may see https://github.com/badilladrian/instrument_elt/blob/develop/settings.py
&
https://github.com/badilladrian/instrument_elt/blob/develop/helpers/utils.py#L5

Reply:
Done, I had this already implemented, skipped adding it here for some reason